whelk-core/src/main/groovy/whelk/component/PostgreSQLComponent.groovy

```diff
@@ -720,20 +720,21 @@ class PostgreSQLComponent {
 
     void reDenormalize() {
         log.info("Re-denormalizing data.")
-        withDbConnection {
-            Connection connection = getMyConnection()
-            boolean leaveCacheAlone = true
+        Connection connection = getOuterConnection()
+        connection.setAutoCommit(false)
+        boolean leaveCacheAlone = true
 
-            long count = 0
-            for (Document doc : loadAll(null, false, null, null)) {
-                refreshDerivativeTables(doc, connection, doc.getDeleted(), leaveCacheAlone)
+        long count = 0
+        for (Document doc : loadAll(null, false, null, null)) {
+            refreshDerivativeTables(doc, connection, doc.getDeleted(), leaveCacheAlone)
 
-                ++count
-                if (count % 500 == 0)
-                    log.info("$count records re-denormalized")
-            }
-            clearEmbellishedCache(connection)
+            ++count
+            if (count % 500 == 0)
+                log.info("$count records re-denormalized")
         }
+        clearEmbellishedCache(connection)
+        connection.commit()
+        connection.close()
```

Comment:
Shouldn't this also be in a finally{}-block? In case something after getOuterConnection() throws?

Reply:
Good point, fixed!